chingu-voyages/v52-tier3-team-38
# Ross/feature/user navigation

# Description
Update UserNavbar to handle dynamic user routing and improve user navigation functionality for resident users.

## Related to the following issue 

- Navigation bar.
- Resident Interface, probably?

## Main changes explained:
- Update UserNavbar.js to replace href links with userId routing
- Add Supabase authentication check to get current user
- Add data check for user data loads

## How to test:
1. Check into current branch
2. Do `npm install` and `npm run dev` to run this PR locally
3. Go to `/login` and log in as a resident user
4. Verify the navbar appears at the bottom of the screen
5. Verify clicking "Profile" navigates to `/user/{your-user-id}/profile`
6. Verify clicking "Submit Request" navigates to `/user/{your-user-id}/submit-request`
7. Verify clicking "Service History" navigates to `/user/{your-user-id}/service-history`
8. Verify "About Us" link works
9. Verify GitHub icon opens project repo in new tab
10. Verify active state (blue highlight) shows on current page's nav item
11. Log out and verify navbar redirects to login page

## Screenshots or videos of changes:

![image](https://github.com/user-attachments/assets/16d948b3-e946-4817-a04a-a051604b8f85)


## Note:
- This PR only updates the navigation functionality
- Actual page content updates will come either in later PR's or this one.
- Navigation routes are set up to use that [userId] param.
